watchtower-plugin/src/retrier.rs

```diff
@@ -996,18 +1023,17 @@ mod tests {
             .is_running());
 
         // Wait for the remaining time and re-check
-        tokio::time::sleep(Duration::from_secs_f64(HALF_API_DELAY + MAX_RUN_TIME)).await;
-        assert!(wt_client
+        wait_until!(wt_client
             .lock()
             .unwrap()
             .get_tower_status(&tower_id)
             .unwrap()
             .is_misbehaving());
 
         // Retriers are wiped every polling interval, so we'll need to wait a bit more to check it
-        tokio::time::sleep(Duration::from_secs(POLLING_TIME)).await;
+        tokio::time::sleep(Duration::from_secs_f64(1.5 * POLLING_TIME as f64)).await;
         assert!(!wt_client.lock().unwrap().retriers.contains_key(&tower_id));
```

Comment:
Can we `wait_until!` the retrier is removed first and assert that it's misbehaving and not the other way around?
```rust
// Wait until the tower is no longer being retried.
wait_until!(wt_client
            .lock()
            .unwrap()
            .get_retrier_status(&tower_id)
            .is_none());

// Retriers are wiped every polling interval, so we'll need to wait a bit more to check it
tokio::time::sleep(Duration::from_secs_f64(1.5 * POLLING_TIME as f64)).await;

// The tower should have a misbehaving status.
assert!(wt_client
            .lock()
            .unwrap()
            .get_tower_status(&tower_id)
            .unwrap()
            .is_misbehaving());
```

Reply:
We may even be able to get rid of the intermediate sleep in this case given the retrier should have already been wiped, wouldn't we?

Reply:
> We may even be able to get rid of the intermediate sleep in this case given the retrier should have already been wiped, wouldn't we?

That's correct. The sleep was guarding against early assertion of `the retrier is not found`, so we can drop it.